routes/conversation.js

```diff
@@ -83,6 +82,26 @@ router.put('/read/:id', auth, async (req, res) => {
   }
 });
 
-router.delete('/delete/:id', (req, res) => {});
+// get all conversations. Given a search parameter (optional)
+router.get('/search/:searchterm', auth, async (req, res) => {
+  const search = req.params.searchterm;
+  const user = req.user.id;
+  try {
+
+    const regex = new RegExp(search, 'i');
+    let conversations = await Conversation.find({
+      'members.name': { $regex: regex },
+      'members._id': user,
+    });
+    if (!conversations) return res.status(404).send('No conversations found');
```

Comment:
Instead of an error code, maybe just return an empty array signifying no conversations were found